db/init.sql

```diff
@@ -14,15 +14,24 @@ CREATE TABLE IF NOT EXISTS category (
     name TEXT NOT NULL
 );
 
+CREATE TABLE IF NOT EXISTS professor (
+    id UUID PRIMARY KEY DEFAULT UUID(),
+    name TEXT NOT NULL
+);
+
 CREATE TABLE IF NOT EXISTS course (
   id UUID PRIMARY KEY DEFAULT UUID(),
   university_id UUID NOT NULL,
   name TEXT NOT NULL,
   description TEXT NOT NULL,
-  professor TEXT NOT NULL,
+  professor_id UUID NOT NULL,
```

Comment:
Here you've set professor_id as UUID but in the code is always handled as a string without checking if it's uuid